src/client/components/configuration/tree/tree.tsx

```diff
@@ -0,0 +1,66 @@
+import * as classnames from 'classnames'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+
+import FileIcon from './file.svg'
+import FolderIconOpen from './folder-open.svg'
+import FolderIcon from './folder.svg'
+import * as style from './style.css'
+
+export interface Node {
+  label: string
+  expanded: boolean
+  leaf: boolean
+  selected: boolean
+  children?: Node[]
+}
+
+export interface TreeProps {
+  data: Node
+  level?: number
+  onClick(node: Node): void
+}
+
+@observer
+export class Tree extends React.Component<TreeProps, any> {
+  constructor(props: any, context: any) {
+    super(props, context)
+    this.onClick = this.onClick.bind(this)
```

Comment:
This is not needed when you use class properties (which you're already doing!), so you can delete this line 🙂 

Normal class method, needs the `.bind` in constructor if you want to pass it directly to something which won't keep the context:

```
onClick(e: MouseEvent) {
  // "this" might change
}
```

Arrow function class property, automatically binds for you:
```
onClick = (e: MouseEvent) => {
  // "this" will always be the class instance
}
```